code/drasil-code/lib/Language/Drasil/Code/Imperative/Descriptions.hs

```diff
@@ -164,34 +165,34 @@ constClassDesc = do
   return $ ccDesc $ filter (flip member (Map.filter (cname ==)
     (clsMap g)) . codeName) (constants $ codeSpec g)
 
--- | Returns a description for the generated function that reads input from a file,
--- if it exists.
+-- | Returns a description for the generated function that reads input from a
+-- file, if it exists.
 inFmtFuncDesc :: GenState Description
 inFmtFuncDesc = do
   g <- get
   let ifDesc False = ""
       ifDesc _ = "Reads input from a file with the given file name"
-  return $ ifDesc $ "get_input" `elem` defList g
+  return $ ifDesc $ GetInput `elem` defList g
 
--- | Returns a description for the generated function that checks input constraints,
--- if it exists.
+-- | Returns a description for the generated function that checks input
+-- constraints, if it exists.
 inConsFuncDesc :: GenState Description
 inConsFuncDesc = do
   g <- get
   pAndS <- physAndSfwrCons
   let icDesc False = ""
       icDesc _ = "Verifies that input values satisfy the " ++ pAndS
-  return $ icDesc $ "input_constraints" `elem` defList g
+  return $ icDesc $ InputConstraints `elem` defList g
```

Comment:
I worry that `ifDesc`, `icDesc` etc are still all "hardcoded" to their type (i.e. `icDesc` to input constraint) so that the abstraction gained via this code change is a small epsilon.